R/blob.R

```diff
@@ -19,48 +27,55 @@ setOldClass("blob")
 #'
 #' as.blob(c("Good morning", "Good evening"))
 blob <- function(...) {
-  new_blob(list(...))
+  x <- list2(...)
+  check_raw_list(x)
+  new_blob(x)
 }
 
-#' @export
-#' @rdname blob
-new_blob <- function(x) {
+check_raw_list <- function(x) {
+  quo <- enquo(x)
   if (!is_raw_list(x)) {
-    stop("`x` must be a list of raw vectors", call. = FALSE)
+    stop("`", as_label(quo), "` must be a list of raw vectors", call. = FALSE)
   }
-  structure(x, class = "blob")
 }
 
 #' @export
 #' @rdname blob
-as.blob <- function(x, ...) {
-  UseMethod("as.blob")
-}
-
-#' @export
-as.blob.blob <- function(x, ...) {
-  x
+new_blob <- function(x = list()) {
+  vec_assert(x, list())
```

Comment:
This already checked for in `new_list_of()` — and that's the right place because it's an invariant of `list_of`, not just `blob`.